tests/python/pants_test/backend/python/tasks/test_pytest_run.py

```diff
@@ -650,6 +653,67 @@ def test_one(self):
     self.assertEqual([1, 2], all_statements)
     self.assertEqual([], not_run_statements)
 
+  def test_coverage_omit_test_sources(self):
+    init_subsystem(Target.Arguments)
+    init_subsystem(SourceRootConfig)
+
+    self.add_to_build_file('src/python/util', 'python_library()\n')
+
+    self.create_file(
+      'src/python/util/math.py',
+      dedent("""
+          def one():  # line 1
+            return 1  # line 2
+        """).strip())
+
+    self.add_to_build_file('src/python/util',
+                           'python_tests(name="tests", dependencies = [":util"])\n')
+
+    self.create_file(
+      'src/python/util/math_test.py',
+      dedent("""
+          import unittest                                            # line 1
+
+          from util import math                                      # line 3
+
+          class MathTestInSameDirectoryAsSource(unittest.TestCase):  # line 5
+            def test_one(self):                                      # line 6
+              self.assertEqual(1, math.one())                        # line 7
+        """).strip())
+
+    test = self.target('src/python/util:tests')
+
+    src_path = os.path.join(self.build_root, 'src/python/util/math.py')
+    test_path = os.path.join(self.build_root, 'src/python/util/math_test.py')
+
+    # First run without omitting the test file.
+    self.assertFalse(os.path.isfile(self.coverage_data_file()))
+    coverage_kwargs = {'coverage': 'auto'}
+    context = self.run_tests(targets=[test], **coverage_kwargs)
+    all_statements, not_run_statements = self.load_coverage_data_for(context, src_path)
+    self.assertEqual([1, 2], all_statements)
+    self.assertEqual([], not_run_statements)
+    all_statements, not_run_statements = self.load_coverage_data_for(context, test_path)
+    self.assertEqual([1, 3, 5, 6, 7], all_statements)
+    self.assertEqual([], not_run_statements)
+
+    os.unlink(self.coverage_data_file())
```

Comment:
Can refactor the below 3 lines to use the more modern `pathlib.Path`:

```python
from pathlib import Path

coverage_data_file = Path(self.coverage_data_file())
coverage_data_file.unlink()
self.assertFalse(coverage_data_file.exists())
```

See https://treyhunner.com/2018/12/why-you-should-be-using-pathlib/ for why we are trying to migrate to `pathlib` instead of `os`.

Reply:
This is copypasta from other parts of the file. Let's make changes like this separately, and uniformly in a file.